### PR TITLE
external: increase an unit of read chunk to avoid a whole reload

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #50752

Problem Summary:

### What changed and how does it work?

But changing `ConcurrentReaderBufferSizePerConc` will improve more speed, will adjust it in future PR.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

before:
```
[2024/02/01 10:30:36.659 +00:00] [INFO] [byte_reader.go:324] ["drop data in closeConcurrentReader"] [reloadCnt=2] [dropBytes=38098261] [curBufIdx=0]
[2024/02/01 10:31:49.387 +00:00] [INFO] [byte_reader.go:324] ["drop data in closeConcurrentReader"] [reloadCnt=2] [dropBytes=1212175701] [curBufIdx=0]
[2024/02/01 10:31:49.387 +00:00] [INFO] [gcs.go:420] ["resetting gcs storage"]
[2024/02/01 10:31:49.388 +00:00] [INFO] [reader.go:62] ["read all data completed"] [takeTime=1m18.018994101s] []
```

after
```
[2024/02/01 10:41:36.142 +00:00] [INFO] [byte_reader.go:323] ["drop data in closeConcurrentReader"] [reloadCnt=1] [dropBytes=349525] [curBufIdx=19]
[2024/02/01 10:42:42.564 +00:00] [INFO] [byte_reader.go:323] ["drop data in closeConcurrentReader"] [reloadCnt=1] [dropBytes=21845] [curBufIdx=579]
[2024/02/01 10:42:42.564 +00:00] [INFO] [gcs.go:420] ["resetting gcs storage"]
[2024/02/01 10:42:42.565 +00:00] [INFO] [reader.go:62] ["read all data completed"] [takeTime=1m10.82428177s] []
```

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by D3Hunter — APPROVED** [review]

File: br/pkg/lightning/backend/external/bench_test.go [hunk 1]
```diff
@@ -1120,3 +1143,55 @@ func TestReadStatFile(t *testing.T) {
...
+	intest.AssertNoError(err)
+	endKey, err := hex.DecodeString("00a00000000000000000")
+	intest.AssertNoError(err)
+	bufPool := membuf.NewPool(
+		membuf.WithBlockNum(0),
+		membuf.WithBlockSize(ConcurrentReaderBufferSizePerConc),
+	)
+	output := &memKVsAndBuffers{}
+	now := time.Now()
+	err = readAllData(ctx, store, dataFiles, statFiles, startKey, endKey, bufPool, output)
```
Comment (wjhuang2016) [inline br/pkg/lightning/backend/external/bench_test.go:1194]: Can you check that reload is called once?

File: br/pkg/lightning/backend/external/engine.go [hunk 2]
```diff
@@ -223,7 +223,9 @@ func getFilesReadConcurrency(
 	startOffs, endOffs := offsets[0], offsets[1]
 	for i := range statsFiles {
 		result[i] = (endOffs[i] - startOffs[i]) / uint64(ConcurrentReaderBufferSizePerConc)
-		result[i] = max(result[i], 1)
+		// let the stat internals cover the [startKey, endKey) since seekPropsOffsets
+		// always return an offset that is less than or equal to the key.
+		result[i] += 1
 		if result[i] > 1 {
```
Comment (wjhuang2016) [inline br/pkg/lightning/backend/external/engine.go:229]: Please refine this condition too.

File: br/pkg/lightning/backend/external/bench_test.go (hunk 1 above)
Comment (lance6716) [reply br/pkg/lightning/backend/external/bench_test.go:1194]: https://github.com/pingcap/tidb/pull/50965/commits/082d741a96cc6bf9b20537e9ee37b032e0198cef

**Comment by lance6716** [comment]
ptal @wjhuang2016

**Pull request merged**